tests/components/statsd/test_init.py

```diff
@@ -1,140 +1,141 @@
 """The tests for the StatsD feeder."""
-import unittest
 from unittest import mock
 
 import pytest
+import statsd as statsdlib
 import voluptuous as vol
 
 import homeassistant.components.statsd as statsd
 from homeassistant.const import EVENT_STATE_CHANGED, STATE_OFF, STATE_ON
 import homeassistant.core as ha
-from homeassistant.setup import setup_component
+from homeassistant.setup import async_setup_component
 
-from tests.common import get_test_home_assistant
+from tests.async_mock import MagicMock
 
 
-class TestStatsd(unittest.TestCase):
-    """Test the StatsD component."""
+@pytest.fixture
+def mock_client():
+    """Pytest fixture for statsd library."""
+    with mock.patch("statsd.StatsClient") as mock_client:
+        yield mock_client.return_value
 
-    def setUp(self):  # pylint: disable=invalid-name
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.tear_down_cleanup)
 
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
+def test_invalid_config():
+    """Test configuration with defaults."""
+    config = {"statsd": {"host1": "host1"}}
 
-    def test_invalid_config(self):
-        """Test configuration with defaults."""
-        config = {"statsd": {"host1": "host1"}}
+    with pytest.raises(vol.Invalid):
+        statsd.CONFIG_SCHEMA(None)
+    with pytest.raises(vol.Invalid):
+        statsd.CONFIG_SCHEMA(config)
 
-        with pytest.raises(vol.Invalid):
-            statsd.CONFIG_SCHEMA(None)
-        with pytest.raises(vol.Invalid):
-            statsd.CONFIG_SCHEMA(config)
 
-    @mock.patch("statsd.StatsClient")
-    def test_statsd_setup_full(self, mock_connection):
-        """Test setup with all data."""
-        config = {"statsd": {"host": "host", "port": 123, "rate": 1, "prefix": "foo"}}
-        self.hass.bus.listen = mock.MagicMock()
-        assert setup_component(self.hass, statsd.DOMAIN, config)
-        assert mock_connection.call_count == 1
-        assert mock_connection.call_args == mock.call(
-            host="host", port=123, prefix="foo"
+async def test_statsd_setup_full(hass):
+    """Test setup with all data."""
+    config = {"statsd": {"host": "host", "port": 123, "rate": 1, "prefix": "foo"}}
+    hass.bus.listen = MagicMock()
+    with mock.patch.object(
+        statsdlib.StatsClient, "__init__", return_value=None
+    ) as mock_init:
```

Comment:
Instead of stubbing `__init__`, why not just init the whole object? 
```suggestion
    with patch("statsdlib.StatsClient") as mock_init:
```

Reply:
Also, we should import `patch` from `tests.async_mock` instead of `mock`, as it handles async code too.